### ISSUE TITLE
OpenTelemetry tracing does not work on 1.9.0

### ISSUE THREAD

**Issue by zhenqxuMSFT** [dapr/dapr#5443]
## In what area(s)?

/area runtime

## What version of Dapr?

1.9.0

## Expected Behavior

Tracing data is collected by otel collector

## Actual Behavior

No tracing data is available. No error log in both otel collector and daprd sidecar

## Steps to Reproduce the Problem
1. Prepare two pods with dapr enabled
2. Invoke service to service call between pods
3. Check the data in destination configured in otel collector

Configuration:
```
  tracing:
    samplingRate: "0"
    zipkin:
      endpointAddress: http://otel-collector.ns.svc.cluster.local:9411/api/v2/spans
```

I switched the daprd version to 1.8.3 and everything works.

**Comment by berndverst** [comment]
@zhenqxuMSFT samplingRate "0" means tracing disabled. Do you still experience this issue if you set samplingRate to the default value "1" ?

**Comment by zhenqxuMSFT** [comment]
Hi @berndverst, oops I pasted a wrong configuration. Here's the configuration on the problematic cluster, the samplingRate is "1" and it's not working. And we just changed the image version of daprd to 1.8.3 and tracing started to work. The configuration itself is unchanged.
```
  tracing:
    samplingRate: "1"
    zipkin:
      endpointAddress: http://otel-collector.ns.svc.cluster.local:9411/api/v2/spans
```

**Comment by amolenk** [comment]
Running into the same problem with the following config and 1.9.2:
```yaml
apiVersion: dapr.io/v1alpha1
kind: Configuration
metadata:
  name: daprConfig
spec:
  tracing:
    samplingRate: "1"
    zipkin:
      endpointAddress: "http://localhost:9411/api/v2/spans"
```

Service invocations using HTTP get logged and are visible in Zipkin. Service invocations using gRPC do not show up. When I downgrade to 1.8.6, both HTTP and gRPC calls show up.

**Comment by ItalyPaleAle** [comment]
Fixed in release-1.9
